carma_cloud_client/src/carma_cloud_client_node.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2022 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "carma_cloud_client/carma_cloud_client_node.hpp"
+
+namespace carma_cloud_client
+{
+  namespace std_ph = std::placeholders;
+
+  CarmaCloudClient::CarmaCloudClient(const rclcpp::NodeOptions &options)
+      : carma_ros2_utils::CarmaLifecycleNode(options)
+  {
+    // Create initial config
+    config_ = Config();
+
+    // Declare parameters
+    // config_.example_param = declare_parameter<std::string>("example_param", config_.example_param);
+  }
+
+  rcl_interfaces::msg::SetParametersResult CarmaCloudClient::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+  {
+    
+    auto error = update_params<std::string>({{"url", config_.url}, 
+                                             {"base_hb", config_.base_hb},
+                                             {"base_req", config_.base_req},
+                                             {"base_ack", config_.base_ack},
+                                             {"port", config_.port},
+                                             {"list", config_.list},
+                                             {"method", config_.method}} , parameters);
+
+    rcl_interfaces::msg::SetParametersResult result;
+
+    result.successful = !error;
+
+    return result;
+  }
+
+  carma_ros2_utils::CallbackReturn CarmaCloudClient::handle_on_configure(const rclcpp_lifecycle::State &)
+  {
+    RCLCPP_INFO_STREAM(this->get_logger(), "CarmaCloudClient trying to configure");
+    // Reset config
+    config_ = Config();
+
+    // Load parameters
+    get_parameter<std::string>("url", config_.url);
+    get_parameter<std::string>("base_hb", config_.base_hb);
```

Comment:
Are these parameters from yaml file or the initial values in the config class?

Reply:
the parameters are updated from what is on the yaml file here

Reply:
Not sure I understand the get_parameter function correctly. Is it automatically pick up value from yaml file that key saying "url"  matches the first argument. Then update the second argument config.url with the value from yaml file?